docs/content/concepts/terminology.md

```diff
@@ -1,100 +1,192 @@
+namespaces in that concrete logical cluster.
 
-A logical cluster is a storage level concept that adds an additional attribute to an object’s identifier on a
-kube-apiserver.  Regular servers identify objects by (group, version, resource, optional namespace, name).  A logical
-cluster enriches an identifier: (group, version, resource, **logical cluster name**, optional namespace, name).
+To a user, a logical cluster appears to be a Kubernetes cluster without all the container orchestration specific
+resources (e.g. Pods, ReplicaSets, Deployments...). It has its own discovery, its own OpenAPI spec, and follows the
+Kubernetes-like constraints about uniqueness of Group-Version-Resource and its behavior. For example, it's not
+possible to have two identical GVRs with different schemas in one logical cluster, but it's possible to have two
+identical GVRs with different schemas in different logical clusters.
+
+However, in the most of cases, you'll never create a logical cluster on its own. Instead, you would use an abstraction,
+such as Workspace, to create logical cluster.
+
+!!! note
+    There could be multiple different models that result in logical clusters being created, with different policies or
+    lifecycle, but Workspace is intended to be the most generic representation of the concept with the broadest possible
+    utility to anyone building control planes.
+
+!!! note
+    Throughout the documentation, terms "workspace" and "logical cluster" might be interchanged. In the most of cases,
+    unless stated otherwise, what applies to a workspace applies to a logical cluster as well, because Workspaces are
+    abstractions built on top of logical clusters.
 
 ## Workspace
 
-A workspace models a set of user-facing APIs for CRUD.  Each workspace is backed by a logical cluster, but not all
-logical clusters may be exposed as workspaces.  Creating a Workspace object results in a logical cluster being available
-via a URL for the client to connect and create resources supported by the APIs in that workspace.  There could be
-multiple different models that result in logical clusters being created, with different policies or lifecycles, but
-Workspace is intended to be the most generic representation of the concept with the broadest possible utility to anyone
-building control planes.
+Workspace is an abstraction used to create and provision logical clusters. Aside from creating the LogicalCluster
+object, the responsibility of a Workspace is to initialize the given logical cluster by installing the required
+resources for the given workspace type, creating initial objects, and more.
+
+Workspaces are managed via the `Workspace` resource. It's a simple resource that mainly contains that workspace name,
+type, and URL.
+
+Workspaces are organized in a multi-root tree structure. The root workspace is created by the kcp server by default
```

Comment:
maybe mention the notation `root:foo:bar`?

Reply:
I came up with this:

> Workspace's full name is based on the hierarchy and the user provided name. For example, if you create a workspace called `bar` in a workspace called `foo` (which is a child of the root workspace), the full workspace names will be:
>
> - `root:foo` for the `foo` workspace
> - `root:foo:bar` for the `bar` workspace

However, aside that it should be checked for correctness, I'm wondering if `workspace's full name` is the term that we want to use.

Reply:
workspace path